src/main/java/racingcar/service/RacingcarService.java

```diff
@@ -0,0 +1,44 @@
+package racingcar.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import racingcar.domain.Car;
+import racingcar.dto.CarPerformance;
+import racingcar.dto.RoundResult;
+import racingcar.util.RandomNumberGenerator;
+
+public class RacingcarService {
+
+    private static final int MIN_OIL_AMOUNT = 0;
+    private static final int MAX_OIL_AMOUNT = 9;
+
+    public RoundResult processRound(List<Car> cars) {
+        List<CarPerformance> carPerformances = new ArrayList<>();
+        for (Car car : cars) {
+            car.goIfOilEnough(decideOilAmount());
+            CarPerformance carPerformance = car.getCurrentPerformance();
+            carPerformances.add(carPerformance);
+        }
+        return new RoundResult(carPerformances);
+    }
+
+    public List<String> pickOutWinners(List<Car> cars) {
+        int maxDistance = Car.findMaxDistance(cars);
+        List<Car> winners =  cars.stream()
+                .filter(car -> car.isSameDistance(maxDistance))
+                .toList();
+        return Car.mapCarsToName(winners);
```

Comment:
아래가 낫지 않았을까요? Car.mapCarsToName 내부에서도 도니까 루프를 두 번 도네요
```suggestion
       return cars.stream()
                .filter(car -> car.isSameDistance(maxDistance))
                .map(car -> car.getName())
                .toList();
```
(요 리뷰보단 일급 컬렉션 쪽 리뷰가 더 중요합니당!)

Reply:
말씀해주신대로 일급 컬렉션으로 리팩토링(여러개의 car를 관리하는 Cars 레코드 생성)했고, 해당 레코드 하위에 `findCarNamesWithSameDistance` 메서드로 이동시켰습니다.